Network/Socket/Syscall.hs

```diff
@@ -178,19 +178,22 @@ accept :: SocketAddress sa => Socket -> IO (Socket, sa)
 accept s = withNewSocketAddress $ \sa sz -> do
      fd <- fdSocket s
 #if defined(mingw32_HOST_OS)
-     new_fd <-
+     (new_fd, addr) <-
         if threaded
-           then with (fromIntegral sz) $ \ ptr_len ->
-                  throwSocketErrorIfMinus1Retry "Network.Socket.accept" $
+           then with (fromIntegral sz) $ \ ptr_len -> do
+                  nfd <- throwSocketErrorIfMinus1Retry "Network.Socket.accept" $
                     c_accept_safe fd sa ptr_len
+                  addr' <- peekSocketAddress sa (Just ptr_len)
+                  return (nfd, addr')
            else do
                 paramData <- c_newAcceptParams fd (fromIntegral sz) sa
                 rc        <- asyncDoProc c_acceptDoProc paramData
                 new_fd'   <- c_acceptNewSock paramData
                 c_free paramData
                 when (rc /= 0) $
                      throwSocketErrorCode "Network.Socket.accept" (fromIntegral rc)
-                return new_fd'
+                addr' <- peekSocketAddress sa Nothing
```

Comment:
In the non-threaded `else` case, why are we able to get away with not specifying the sockaddr size?  Is this just a branch that is not reached by the code that needs this PR, or is the issue somehow not applicable here?  Is `c_acceptNewSock` doing something differently?

Reply:
One was never retrieved in the first place, so my edit was to persist the existing functionality cleanly. I don't know what the intended semantics are for this branch, but any change to its behavior should have a test environment to ensure the "correct" thing is happening.